include/cuco/static_map.cuh

```diff
@@ -296,6 +296,26 @@ class static_map {
             KeyEqual key_equal  = KeyEqual{},
             cudaStream_t stream = 0);
 
+  /**
+   * @brief Retrieves all of the keys and their associated values.
+   *
+   * The order in which keys are returned is implementation defined and not guaranteed to be
+   * consistent between subsequent calls to `retrieve_all`.
+   *
+   * Behavior is undefined if the range beginning at `keys_out` or `values_out` is not large enough
+   * to contain the number of keys in the map.
+   *
+   * @tparam KeyOut Device accessible random access output iterator whose `value_type` is
+   * convertible from `key_type`.
+   * @tparam ValueOut Device accesible random access output iterator whose `value_type` is
+   * convertible from `mapped_type`.
+   * @param keys_out Beginning output iterator for keys
+   * @param values_out Beginning output iterator for values
+   * @param stream CUDA stream used for this operation
+   */
+  template <typename KeyOut, typename ValueOut>
+  void retrieve_all(KeyOut keys_out, ValueOut values_out, cudaStream_t stream = 0);
```

Comment:
This should either return the number of retrieved keys or return a `pair<KeyOut, ValueOut>` iterators to the end of the two output ranges.

Reply:
Update API to return a pair of iterators pointing to the end of the output.